docs/upgrades.md

```diff
@@ -0,0 +1,58 @@
+## Agent Upgrades
+
+### Communications amongst components
+The following sequence diagram illustrates the process of upgrading a
+Fleet-managed Agent. The diagram focusses on the communications that occur
+amongst the various components involved in the upgrade process.
+
+This diagram is accurate as of version `8.9.0` of every component shown.
+
+```mermaid
+sequenceDiagram
+    actor U as User
+    participant UI as Fleet UI
+    participant ES
+    participant FS as Fleet Server
+    participant A as Agent
+    participant UW as Upgrade Watcher
+    participant UM as Upgrader Marker
+
+    U->>UI: Initiate upgrade
+    UI->>ES: Update Agent doc in `.fleet-agents`<br />set `upgrade_started_at`
+    UI->>UI: Show Agent status as "updating"
+    UI->>ES: Create new doc in `.fleet-actions` for `UPGRADE` action
+    A->>FS: Check-in request
+    FS->>ES: Read pending actions from .fleet-actions
+    FS->>A: Check-in response
+    A->>A: Queue upgrade action
+    alt If upgrade start fails
+       A->>FS: Ack failed upgrade
+       FS->>ES: Update Agent doc in `.fleet-agents`<br />set `upgrade_status` = "failed"
+       UI->>UI: Show Agent status as "???"
+       Note right of UI: Need to check
```

Comment:
TODO: check and remove note.

Reply:
Yeah I think the agent status stays in updating.  I think in some cases the error is written to `.fleet-action-results` and the UI shows the action failure in Agent activity flyout.